overlord/configstate/configcore/debug.go

```diff
@@ -0,0 +1,166 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//go:build !nomanagers
+
+/*
+ * Copyright (C) 2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package configcore
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/systemd"
+)
+
+const (
+	optionDebugSnapdLog            = "debug.snapd.log"
+	optionDebugSystemdLogLevel     = "debug.systemd.log-level"
+	coreOptionDebugSnapdLog        = "core." + optionDebugSnapdLog
+	coreOptionDebugSystemdLogLevel = "core." + optionDebugSystemdLogLevel
+)
+
+func init() {
+	supportedConfigurations[coreOptionDebugSnapdLog] = true
+	supportedConfigurations[coreOptionDebugSystemdLogLevel] = true
+}
+
+func optionChanged(tr RunTransaction, fullOptionName string) bool {
+	for _, name := range tr.Changes() {
+		if name == fullOptionName {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
```suggestion
```

We have `strutil.ListContains` which does something similar, so we could use that here